test/libsolidity/SolidityNameAndTypeResolution.cpp

```diff
@@ -3750,6 +3750,31 @@ BOOST_AUTO_TEST_CASE(one_divided_by_three_integer_conversion)
 	BOOST_CHECK(!success(text));
 }
 
+BOOST_AUTO_TEST_CASE(unused_return_value)
+{
+	char const* text = R"(
+		contract test {
+			function g() returns (uint) {}
+			function f() {
+				g();
+			}
+		}
+	)";
+	BOOST_CHECK(expectError(text, true) == Error::Type::Warning);
+}
+
+BOOST_AUTO_TEST_CASE(unused_return_value_send)
+{
+	char const* text = R"(
+		contract test {
+			function f() {
+				address(0x12).send(1);
```

Comment:
Sorry if what I'm about to say is obvious. Bear with me as I'm learning lots at once. After a quick review of the code (disclaimer: I never learned C/C++ very well), it looks like this is part of the visitor pattern. If that's the case, then it looks like this code is being run on every single `ExpressionStatement` in the AST. So I think that means scope is irrelevant.